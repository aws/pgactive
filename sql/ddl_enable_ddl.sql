SET bdr.permit_ddl_locking = false;
CREATE TABLE should_fail ( id integer );

RESET bdr.permit_ddl_locking;
CREATE TABLE create_ok (id integer);

SET bdr.permit_ddl_locking = false;
ALTER TABLE create_ok ADD COLUMN alter_should_fail text;

RESET bdr.permit_ddl_locking;
DROP TABLE create_ok;

-- Now for the rest of the DDL tests, presume they're allowed,
-- otherwise they'll get pointlessly verbose.
ALTER DATABASE regression RESET bdr.permit_ddl_locking;
ALTER DATABASE postgres RESET bdr.permit_ddl_locking;
