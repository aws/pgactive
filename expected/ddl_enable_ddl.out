SET bdr.permit_ddl_locking = false;
CREATE TABLE should_fail ( id integer );
ERROR:  global DDL locking attempt rejected by configuration
DETAIL:  bdr.permit_ddl_locking is false and the attempted command would require the global lock to be acquired. Command rejected.
HINT:  See the 'DDL replication' chapter of the documentation.
RESET bdr.permit_ddl_locking;
CREATE TABLE create_ok (id integer);
SET bdr.permit_ddl_locking = false;
ALTER TABLE create_ok ADD COLUMN alter_should_fail text;
ERROR:  global DDL locking attempt rejected by configuration
DETAIL:  bdr.permit_ddl_locking is false and the attempted command would require the global lock to be acquired. Command rejected.
HINT:  See the 'DDL replication' chapter of the documentation.
RESET bdr.permit_ddl_locking;
DROP TABLE create_ok;
-- Now for the rest of the DDL tests, presume they're allowed,
-- otherwise they'll get pointlessly verbose.
ALTER DATABASE regression RESET bdr.permit_ddl_locking;
ALTER DATABASE postgres RESET bdr.permit_ddl_locking;
